File: src/mock/purchasing.ts
import type {
  PurchaseOrderDetail,
  PurchaseOrderItem,
  PurchasingDashboardData,
  PurchasingSummary
} from "@/types/purchasing";

export const purchasingSummary: PurchasingSummary[] = [
  {
    label: "未收採購單",
    value: "12",
    hint: "到期或逾期 3 筆",
    tone: "warning"
  },
  {
    label: "交期風險",
    value: "2 / 4",
    hint: "逾期 2 筆，影響工單 2 筆",
    tone: "danger"
  },
  {
    label: "採購金額",
    value: "$3,642,000",
    hint: "查詢期間進貨單 18 筆",
    tone: "info"
  },
  {
    label: "未連請購",
    value: "5",
    hint: "待入庫交接 6 筆",
    tone: "warning"
  }
];

export const purchaseOrders: PurchaseOrderItem[] = [
  {
    id: "PO-20260724-031",
    purchaseOrderNo: "PO-20260724-031",
    purchaseDate: "2026/7/24",
    purchaseDateTimestamp: 1784822400,
    itemNo: "RM-CORN-001",
    itemName: "冷凍玉米粒",
    unit: "公斤",
    unitCode: 2,
    supplierNo: "SUP-GREEN-01",
    supplierName: "綠田食品",
    orderedCount: 1200,
    receivedCount: 600,
    openCount: 600,
    unitPrice: 72,
    purchaseAmount: 86400,
    expectedArrivalDate: "2026/7/28",
    expectedArrivalTimestamp: 1785168000,
    purchaseRequestNo: "PR-20260723-018",
    purchaseRequestLinkStatusCode: "linked",
    purchaseRequestLinkStatus: "已連請購",
    sourceOrderNo: "SO-20260723-022",
    linkedWorkOrderNo: "WO-20260728-002",
    warehouseStatusCode: "pending_putaway",
    warehouseStatus: "待入庫交接",
    riskLevelCode: 3,
    riskLevel: "高風險",
    riskType: "late_arrival",
    riskTypeLabel: "逾期未到",
    tone: "danger"
  },
  {
    id: "PO-20260725-014",
    purchaseOrderNo: "PO-20260725-014",
    purchaseDate: "2026/7/25",
    purchaseDateTimestamp: 1784908800,
    itemNo: "PK-BAG-010",
    itemName: "耐熱殺菌袋",
    unit: "個",
    unitCode: 101,
    supplierNo: "SUP-PACK-02",
    supplierName: "台灣包材",
    orderedCount: 20000,
    receivedCount: 0,
    openCount: 20000,
    unitPrice: 6.6,
    purchaseAmount: 132000,
    expectedArrivalDate: "2026/7/30",
    expectedArrivalTimestamp: 1785340800,
    purchaseRequestNo: "",
    purchaseRequestLinkStatusCode: "unlinked",
    purchaseRequestLinkStatus: "未連請購",
    sourceOrderNo: "",
    linkedWorkOrderNo: "WO-20260731-006",
    warehouseStatusCode: "not_received",
    warehouseStatus: "尚未到貨",
    riskLevelCode: 1,
    riskLevel: "注意",
    riskType: "purchase_request_unlinked",
    riskTypeLabel: "缺少請購關聯",
    tone: "warning"
  },
  {
    id: "PO-20260726-027",
    purchaseOrderNo: "PO-20260726-027",
    purchaseDate: "2026/7/26",
    purchaseDateTimestamp: 1784995200,
    itemNo: "RM-CHICKEN-022",
    itemName: "雞胸肉原料",
    unit: "公斤",
    unitCode: 2,
    supplierNo: "SUP-POULTRY-03",
    supplierName: "安心禽品",
    orderedCount: 1600,
    receivedCount: 1600,
    openCount: 0,
    unitPrice: 110,
    purchaseAmount: 176000,
    expectedArrivalDate: "2026/7/29",
    expectedArrivalTimestamp: 1785254400,
    purchaseRequestNo: "PR-20260725-006",
    purchaseRequestLinkStatusCode: "linked",
    purchaseRequestLinkStatus: "已連請購",
    sourceOrderNo: "SO-20260726-018",
    linkedWorkOrderNo: "WO-20260729-001",
    warehouseStatusCode: "stocked",
    warehouseStatus: "已入庫",
    riskLevelCode: 0,
    riskLevel: "正常",
    riskType: "normal",
    riskTypeLabel: "正常",
    tone: "success"
  }
];

export const purchasingDashboardMock: PurchasingDashboardData & { details: Record<string, PurchaseOrderDetail> } = {
  range: {
    startDate: "2026-07-01",
    endDate: "2026-07-29",
    startTimestamp: 1782835200,
    endTimestamp: 1785340799
  },
  summary: purchasingSummary,
  purchaseOrders,
  deliveryRisks: [
    {
      ...purchaseOrders[0],
      shortageCount: 600,
      shortageValue: 43200,
      impactSourceType: "work_order",
      impactSourceNo: "WO-20260728-002",
      impactSourceLabel: "生產工單",
      followUpCode: "confirm_supplier_date",
      followUpLabel: "確認供應商交期"
    },
    {
      ...purchaseOrders[1],
      shortageCount: 20000,
      shortageValue: 132000,
      impactSourceType: "safety_stock",
      impactSourceNo: "PK-BAG-010",
      impactSourceLabel: "安全水位",
      followUpCode: "review_source_impact",
      followUpLabel: "檢視來源影響"
    }
  ],
  receipts: [
    {
      id: "GRN-20260728-018",
      no: "GRN-20260728-018",
      purchaseOrderNo: "PO-20260724-031",
      date: "2026/7/28",
      dateTimestamp: 1785168000,
      category: 0,
      categoryLabel: "進貨",
      itemNo: "RM-CORN-001",
      itemName: "冷凍玉米粒",
      expectedCount: 600,
      checkedCount: 600,
      receivedCount: 600,
      receivingStatusCode: "received",
      receivingStatus: "已收貨",
      warehouseStatusCode: "pending_putaway",
      warehouseStatus: "待入庫交接",
      nextOwnerDepartment: 4,
      nextOwnerDepartmentLabel: "倉庫"
    },
    {
      id: "GRN-20260729-021",
      no: "GRN-20260729-021",
      purchaseOrderNo: "PO-20260726-027",
      date: "2026/7/29",
      dateTimestamp: 1785254400,
      category: 0,
      categoryLabel: "進貨",
      itemNo: "RM-CHICKEN-022",
      itemName: "雞胸肉原料",
      expectedCount: 1600,
      checkedCount: 1600,
      receivedCount: 1600,
      receivingStatusCode: "received",
      receivingStatus: "已收貨",
      warehouseStatusCode: "stocked",
      warehouseStatus: "已入庫",
      nextOwnerDepartment: 0,
      nextOwnerDepartmentLabel: "未指定"
    }
  ],
  suppliers: [
    {
      id: "SUP-GREEN-01",
      supplierNo: "SUP-GREEN-01",
      supplierName: "綠田食品",
      purchaseOrderCount: 7,
      openPurchaseOrderCount: 3,
      latePurchaseOrderCount: 1,
      purchaseAmount: 942000,
      pendingReceiptCount: 1800,
      riskLevelCode: 3,
      riskLevel: "高風險",
      tone: "danger"
    },
    {
      id: "SUP-PACK-02",
      supplierNo: "SUP-PACK-02",
      supplierName: "台灣包材",
      purchaseOrderCount: 5,
      openPurchaseOrderCount: 4,
      latePurchaseOrderCount: 0,
      purchaseAmount: 626000,
      pendingReceiptCount: 28000,
      riskLevelCode: 1,
      riskLevel: "注意",
      tone: "warning"
    }
  ],
  total: {
    purchaseOrders: 3,
    deliveryRisks: 2,
    receipts: 2,
    suppliers: 2
  },
  details: {
    "PO-20260724-031": {
      purchaseOrderNo: "PO-20260724-031",
      purchaseDate: "2026/7/24",
      itemNo: "RM-CORN-001",
      itemName: "冷凍玉米粒",
      unit: "公斤",
      supplierName: "綠田食品",
      orderedCount: 1200,
      unitPrice: 72,
      purchaseAmount: 86400,
      expectedArrivalDate: "2026/7/28",
      comment: "分批到貨，第二批需供應商確認車次。",
      purchaseRequestNo: "PR-20260723-018",
      sourceOrderNo: "SO-20260723-022",
      linkedWorkOrderNo: "WO-20260728-002",
      inventory: { currentCount: 600, reservedCount: 420, availableCount: 180 },
      receipts: [
        {
          no: "GRN-20260728-018",
          date: "2026/7/28",
          categoryLabel: "進貨",
          expectedCount: 600,
          checkedCount: 600,
          receivedCount: 600,
          receivingStatus: "已收貨",
          warehouseStatus: "待入庫交接"
        }
      ],
      workflow: [
        {
          taskId: "WF-PO-001",
          taskTypeLabel: "採購",
          refNo: "PO-20260724-031",
          taskStatusLabel: "已完成",
          ownerDepartmentLabel: "採購",
          tone: "success"
        },
        {
          taskId: "WF-GRN-018",
          taskTypeLabel: "入庫",
          refNo: "GRN-20260728-018",
          taskStatusLabel: "待處理",
          ownerDepartmentLabel: "倉庫",
          tone: "warning"
        }
      ],
      relatedDocuments: { quoteNo: "QT-20260718-004", contractNo: "CT-20260701-009" }
    }
  }
};

File: restserver/package/restserver/api/v2/purchasing.py
# coding=utf8
import time
from collections import defaultdict

from flask import request
from sqlalchemy import or_

from package.common.common import (
    EGoodsReceiptNoteCategory,
    EPurchasingRiskType,
    EProductionRiskLevel,
)
from package.dbwrapper.dbmgr import CDBMgr
from package.dbwrapper.table import (
    CTableCompany,
    CTableGoodsReceiptNote,
    CTablePurchaseOrder,
    CTablePurchaseRequest,
)
from package.util.util import (
    util_build_local_date_range, util_round_amount, util_round_price, util_round_quantity,
    util_safe_float, util_safe_int,
)


class CPurchasingPurchaseOrderService(object):
    def __init__(self, str_timezone):
        self.str_timezone = str_timezone or "UTC"

    def __range(self):
        str_start = request.args.get("startDate", "", type=str)
        str_end = request.args.get("endDate", "", type=str)
        if not str_start or not str_end:
            return None
        return util_build_local_date_range(str_start, str_end, self.str_timezone)

    def __base_orders(self, obj_session, dict_range):
        str_keyword = request.args.get("keyword", "", type=str)
        str_supplier_no = request.args.get("supplierNo", "", type=str)
        lst_filter = [
            CTablePurchaseOrder.date >= dict_range["startTimestamp"],
            CTablePurchaseOrder.date <= dict_range["endTimestamp"],
        ]
        if str_supplier_no:
            lst_filter.append(CTablePurchaseOrder.item_ref_no == str_supplier_no)
        if str_keyword:
            str_like = "%" + str_keyword + "%"
            lst_filter.append(or_(CTablePurchaseOrder.no.like(str_like),
                                  CTablePurchaseOrder.item_no.like(str_like),
                                  CTablePurchaseOrder.item_name.like(str_like)))
        return obj_session.query(CTablePurchaseOrder).filter(*lst_filter).order_by(
            CTablePurchaseOrder.expectedDate.asc(), CTablePurchaseOrder.no.asc()
        ).all()

    def __receipts(self, obj_session, lst_order_no):
        if not lst_order_no:
            return defaultdict(list)
        dict_receipts = defaultdict(list)
        for obj_row in obj_session.query(CTableGoodsReceiptNote).filter(
                CTableGoodsReceiptNote.purchase_order_no.in_(lst_order_no)).order_by(
                CTableGoodsReceiptNote.date.asc(), CTableGoodsReceiptNote.no.asc()).all():
            dict_receipts[obj_row.purchase_order_no].append(obj_row)
        return dict_receipts

    def __received_count(self, lst_receipts):
        return util_round_quantity(sum(
            util_safe_float(obj_row.checkedCount) * (-1 if obj_row.category == EGoodsReceiptNoteCategory.RETURN else 1)
            for obj_row in lst_receipts
        ))

    def __row(self, obj_session, obj_order, dict_receipts, n_now):
        obj_company = obj_session.query(CTableCompany).filter(
            CTableCompany.no == obj_order.item_ref_no).first()
        obj_request = None
        if obj_order.purchase_request_no:
            obj_request = obj_session.query(CTablePurchaseRequest).filter(
                CTablePurchaseRequest.no == obj_order.purchase_request_no).first()
        f_ordered = util_round_quantity(obj_order.count)
        f_received = self.__received_count(dict_receipts.get(obj_order.no, []))
        f_open = util_round_quantity(max(f_ordered - f_received, 0))
        n_expected = util_safe_int(obj_order.expectedDate)
        n_risk_level = EProductionRiskLevel.NORMAL
        str_risk_type = EPurchasingRiskType.NORMAL
        if f_open > 0:
            if n_expected and n_expected < n_now:
                n_risk_level = EProductionRiskLevel.DANGER
                str_risk_type = EPurchasingRiskType.LATE_ARRIVAL
            elif n_expected and util_build_local_date_range(
                    time.strftime("%Y-%m-%d", time.localtime(n_now)),
                    time.strftime("%Y-%m-%d", time.localtime(n_now)), self.str_timezone
            ) and n_expected <= n_now + 86400:
                n_risk_level = EProductionRiskLevel.NOTICE
                str_risk_type = EPurchasingRiskType.DUE_TODAY
            elif not obj_request:
                n_risk_level = EProductionRiskLevel.NOTICE
                str_risk_type = EPurchasingRiskType.PURCHASE_REQUEST_UNLINKED
            else:
                n_risk_level = EProductionRiskLevel.NOTICE
                str_risk_type = EPurchasingRiskType.OPEN_RECEIPT
        return {
            "purchaseOrderNo": obj_order.no or "", "purchaseDateTimestamp": util_safe_int(obj_order.date),
            "itemNo": obj_order.item_no or "", "itemName": obj_order.item_name or "", "unit": util_safe_int(obj_order.unit),
            "supplierNo": obj_order.item_ref_no or "", "supplierName": getattr(obj_company, "displayName", "") or "",
            "orderedCount": f_ordered, "receivedCount": f_received, "openCount": f_open,
            "unitPrice": util_round_price(obj_order.price), "purchaseAmount": util_round_amount(obj_order.amount),
            "expectedArrivalTimestamp": n_expected, "purchaseRequestNo": obj_order.purchase_request_no or "",
            "purchaseRequestLinkStatusCode": "linked" if obj_request else ("unlinked" if not obj_order.purchase_request_no else "invalid"),
            "sourceOrderNo": getattr(obj_request, "product_order_no", "") or "", "linkedWorkOrderNo": "",
            "warehouseStatusCode": "not_received" if not dict_receipts.get(obj_order.no) else "unknown",
            "riskLevel": n_risk_level, "riskType": str_risk_type,
        }

    def __response(self, dict_range, dict_payload):
        return {"serverTimestamp": util_safe_int(time.time()), "timezone": self.str_timezone,
                "range": dict_range, **dict_payload}

    def get_dashboard(self):
        dict_range = self.__range()
        if not dict_range:
            return None
        with CDBMgr() as obj_dbmgr:
            obj_session = obj_dbmgr.get_session()
            lst_orders = self.__base_orders(obj_session, dict_range)
            dict_receipts = self.__receipts(obj_session, [obj_row.no for obj_row in lst_orders])
            lst_rows = [self.__row(obj_session, obj_row, dict_receipts, int(time.time())) for obj_row in lst_orders]
            n_risk_level = request.args.get("riskLevel", None, type=int)
            if n_risk_level is not None:
                lst_rows = [dict_row for dict_row in lst_rows if dict_row["riskLevel"] == n_risk_level]
            n_start = max(request.args.get("start", 0, type=int), 0)
            n_count = min(max(request.args.get("count", 50, type=int), 1), 100)
            return self.__response(dict_range, {"summary": {
                "openPurchaseOrderCount": sum(1 for dict_row in lst_rows if dict_row["openCount"] > 0),
                "lateOrDueTodayCount": sum(1 for dict_row in lst_rows if dict_row["riskType"] in (
                    EPurchasingRiskType.LATE_ARRIVAL, EPurchasingRiskType.DUE_TODAY)),
                "purchaseAmount": util_round_amount(sum(dict_row["purchaseAmount"] for dict_row in lst_rows)),
                "unlinkedPurchaseRequestCount": sum(1 for dict_row in lst_rows if dict_row["purchaseRequestLinkStatusCode"] != "linked"),
            }, "items": lst_rows[n_start:n_start + n_count], "total": len(lst_rows), "start": n_start,
                "count": len(lst_rows[n_start:n_start + n_count])})

    def get_delivery_risk(self):
        dict_payload = self.get_dashboard()
        if dict_payload is None:
            return None
        lst_rows = [dict_row for dict_row in dict_payload["items"] if dict_row["riskLevel"] != EProductionRiskLevel.NORMAL]
        dict_payload["items"] = lst_rows
        dict_payload["total"] = len(lst_rows)
        dict_payload["count"] = len(lst_rows)
        dict_payload["summary"] = {
            "highRiskCount": sum(1 for dict_row in lst_rows if dict_row["riskLevel"] == EProductionRiskLevel.DANGER),
            "noticeCount": sum(1 for dict_row in lst_rows if dict_row["riskLevel"] == EProductionRiskLevel.NOTICE),
            "lateCount": sum(1 for dict_row in lst_rows if dict_row["riskType"] == EPurchasingRiskType.LATE_ARRIVAL),
            "affectedWorkOrderCount": 0, "averageLateDays": 0.0,
        }
        for dict_row in dict_payload["items"]:
            f_shortage = dict_row.pop("openCount")
            dict_row.update({"shortageCount": f_shortage,
                             "shortageValue": util_round_amount(f_shortage * dict_row["unitPrice"]),
                             "impactSourceType": "unknown", "impactSourceNo": "",
                             "followUpCode": "confirm_supplier_date"})
        return dict_payload

    def get_receipts(self):
        dict_range = self.__range()
        if not dict_range:
            return None
        with CDBMgr() as obj_dbmgr:
            obj_session = obj_dbmgr.get_session()
            lst_rows = obj_session.query(CTableGoodsReceiptNote).filter(
                CTableGoodsReceiptNote.date >= dict_range["startTimestamp"],
                CTableGoodsReceiptNote.date <= dict_range["endTimestamp"],
            ).order_by(CTableGoodsReceiptNote.date.asc(), CTableGoodsReceiptNote.no.asc()).all()
            lst_result = []
            for obj_row in lst_rows:
                str_receiving = "returned" if obj_row.category == 1 else ("received" if util_safe_float(obj_row.checkedCount) > 0 else "unknown")
                lst_result.append({"no": obj_row.no or "", "purchaseOrderNo": obj_row.purchase_order_no or "",
                    "dateTimestamp": util_safe_int(obj_row.date), "category": util_safe_int(obj_row.category),
                    "itemNo": obj_row.item_no or "", "itemName": obj_row.item_name or "",
                    "expectedCount": util_round_quantity(obj_row.expectedCount), "checkedCount": util_round_quantity(obj_row.checkedCount),
                    "receivedCount": util_round_quantity(obj_row.checkedCount), "receivingStatusCode": str_receiving,
                    "warehouseStatusCode": "unknown", "nextOwnerDepartment": 0})
            n_start = max(request.args.get("start", 0, type=int), 0)
            n_count = min(max(request.args.get("count", 50, type=int), 1), 100)
            return self.__response(dict_range, {"summary": {"receiptCount": len(lst_result), "pendingPutawayCount": len(lst_result)},
                "items": lst_result[n_start:n_start + n_count], "total": len(lst_result), "start": n_start,
                "count": len(lst_result[n_start:n_start + n_count])})

    def get_suppliers(self):
        dict_payload = self.get_dashboard()
        if dict_payload is None:
            return None
        dict_rows = {}
        for dict_row in dict_payload["items"]:
            str_key = dict_row["supplierNo"]
            dict_item = dict_rows.setdefault(str_key, {"supplierNo": str_key, "supplierName": dict_row["supplierName"],
                "purchaseOrderCount": 0, "openPurchaseOrderCount": 0, "latePurchaseOrderCount": 0,
                "purchaseAmount": 0, "pendingReceiptCount": 0,
                "riskLevel": EProductionRiskLevel.NORMAL})
            dict_item["purchaseOrderCount"] += 1
            dict_item["openPurchaseOrderCount"] += int(dict_row["openCount"] > 0)
            dict_item["latePurchaseOrderCount"] += int(
                dict_row["riskType"] == EPurchasingRiskType.LATE_ARRIVAL
            )
            dict_item["purchaseAmount"] += dict_row["purchaseAmount"]
            dict_item["pendingReceiptCount"] += dict_row["openCount"]
            if dict_row["riskLevel"] > dict_item["riskLevel"]:
                dict_item["riskLevel"] = dict_row["riskLevel"]
        lst_result = list(dict_rows.values())
        for dict_row in lst_result:
            dict_row["purchaseAmount"] = util_round_amount(dict_row["purchaseAmount"])
            dict_row["pendingReceiptCount"] = util_round_quantity(dict_row["pendingReceiptCount"])
        dict_payload["items"] = lst_result
        dict_payload["total"] = len(lst_result)
        dict_payload["count"] = len(lst_result)
        return dict_payload

    def get_detail(self, str_order_no):
        with CDBMgr() as obj_dbmgr:
            obj_session = obj_dbmgr.get_session()
            obj_order = obj_session.query(CTablePurchaseOrder).filter(CTablePurchaseOrder.no == str_order_no).first()
            if not obj_order:
                return None
            obj_company = obj_session.query(CTableCompany).filter(CTableCompany.no == obj_order.item_ref_no).first()
            obj_request = obj_session.query(CTablePurchaseRequest).filter(CTablePurchaseRequest.no == obj_order.purchase_request_no).first() if obj_order.purchase_request_no else None
            lst_receipts = obj_session.query(CTableGoodsReceiptNote).filter(CTableGoodsReceiptNote.purchase_order_no == str_order_no).order_by(CTableGoodsReceiptNote.date.asc(), CTableGoodsReceiptNote.no.asc()).all()
            return {"serverTimestamp": util_safe_int(time.time()), "timezone": self.str_timezone,
                "purchaseOrder": {"purchaseOrderNo": obj_order.no or "", "purchaseDateTimestamp": util_safe_int(obj_order.date), "itemNo": obj_order.item_no or "", "itemName": obj_order.item_name or "", "unit": util_safe_int(obj_order.unit), "supplierNo": obj_order.item_ref_no or "", "supplierName": getattr(obj_company, "displayName", "") or "", "orderedCount": util_round_quantity(obj_order.count), "unitPrice": util_round_price(obj_order.price), "purchaseAmount": util_round_amount(obj_order.amount), "expectedArrivalTimestamp": util_safe_int(obj_order.expectedDate), "comment": obj_order.comment or ""},
                "purchaseRequest": {"purchaseRequestNo": obj_request.no, "sourceOrderNo": obj_request.product_order_no or "", "itemNo": obj_request.item_no or "", "requestedCount": util_round_quantity(obj_request.count)} if obj_request else None,
                "supplier": {"supplierNo": obj_order.item_ref_no or "", "supplierName": getattr(obj_company, "displayName", "") or ""},
                "receipts": [{"no": obj_row.no or "", "dateTimestamp": util_safe_int(obj_row.date), "category": util_safe_int(obj_row.category), "expectedCount": util_round_quantity(obj_row.expectedCount), "checkedCount": util_round_quantity(obj_row.checkedCount), "receivedCount": util_round_quantity(obj_row.checkedCount), "receivingStatusCode": "returned" if obj_row.category == 1 else ("received" if util_safe_float(obj_row.checkedCount) > 0 else "unknown"), "warehouseStatusCode": "unknown"} for obj_row in lst_receipts],
                "source": {"sourceOrderNo": getattr(obj_request, "product_order_no", "") or "", "linkedWorkOrderNo": ""},
                "inventory": {"currentCount": 0.0, "reservedCount": 0.0, "availableCount": 0.0}, "workflow": [],
                "relatedDocuments": {"quoteNo": "", "contractNo": ""}}


class CPurchasingPurchaseOrderDashboard(object):
    def get(self, str_timezone, str_id):
        obj_service = CPurchasingPurchaseOrderService(str_timezone)
        dict_payload = obj_service.get_dashboard()
        return self._response(dict_payload)

    @staticmethod
    def _response(dict_payload):
        if dict_payload is None:
            return 400, 1, "invalid startDate/endDate or record not found", {}
        return 200, 0, "success", dict_payload


class CPurchasingPurchaseOrderDeliveryRisk(CPurchasingPurchaseOrderDashboard):
    def get(self, str_timezone, str_id):
        return self._response(CPurchasingPurchaseOrderService(str_timezone).get_delivery_risk())


class CPurchasingGoodsReceiptDashboard(CPurchasingPurchaseOrderDashboard):
    def get(self, str_timezone, str_id):
        return self._response(CPurchasingPurchaseOrderService(str_timezone).get_receipts())


class CPurchasingSupplierDashboard(CPurchasingPurchaseOrderDashboard):
    def get(self, str_timezone, str_id):
        return self._response(CPurchasingPurchaseOrderService(str_timezone).get_suppliers())


class CPurchasingPurchaseOrderDetail(CPurchasingPurchaseOrderDashboard):
    def get(self, str_timezone, str_id):
        return self._response(CPurchasingPurchaseOrderService(str_timezone).get_detail(str_id))
